teerex/src/lib.rs

```diff
@@ -258,9 +285,15 @@ pub mod pallet {
 					latest_block_header,
 				);
 				latest_block_number = latest_block_header.block_number;
-				while <SidechainBlockHeaderQueue<T>>::contains_key(latest_block_number + 1) {
-					let header = <SidechainBlockHeaderQueue<T>>::take(latest_block_number + 1);
-					<SidechainBlockHeaderQueue<T>>::remove(latest_block_number + 1);
+				let mut i: u64 = 0;
+				while <SidechainBlockHeaderQueue<T>>::contains_key(
+					shard_id,
+					latest_block_number + 1,
+				) && i < early_block_proposal_lenience
+				{
+					let header =
+						<SidechainBlockHeaderQueue<T>>::take(shard_id, latest_block_number + 1);
```

Comment:
I mean theoretically, we checked before that `latest_block_number + 1` does not overflow, but I'd still like to see the checked operations. Such a pattern is not safe when aposterio code changes remove the checked math operatios a few lines above.

Anyhow, here we could just allocate a variable `next_block_number` to produce less overhead code.

Reply:
I created a function to compare with the blocknumber, since the code is also very similar to the lenience comparison.